src/rules/restrictPlusOperandsRule.ts

```diff
@@ -60,12 +63,34 @@ class RestrictPlusOperandsWalker extends Lint.ProgramAwareRuleWalker {
     }
 }
 
-function getAddableType(node: ts.Expression, tc: ts.TypeChecker): "string" | "number" | "invalid" {
+function getAddableType(node: ts.Expression, tc: ts.TypeChecker): AddTypes {
```

Comment:
I don't think this extra function in the call stack is necessary -- would prefer to inline usage of `getBaseTypeOfLiteralType` above, like so:

```ts
const tc = this.getTypeChecker();
const leftType = getBaseTypeOfLiteralType(ts.getTypeAtLocation(node.left));
...
```